server/src/internalClusterTest/java/org/elasticsearch/cluster/routing/allocation/decider/WriteLoadConstraintDeciderIT.java

```diff
@@ -849,14 +803,16 @@ private TestHarness setUpThreeTestNodesAndAllIndexShardsOnFirstNode() {
             );
         });
 
-        createIndex(
-            indexName,
-            Settings.builder().put(SETTING_NUMBER_OF_SHARDS, randomNumberOfShards).put(SETTING_NUMBER_OF_REPLICAS, 0).build()
-        );
+        createIndex(indexName, randomNumberOfShards, 0);
         ensureGreen(indexName);
 
-        logger.info("---> Waiting for all [" + randomNumberOfShards + "] shards to be assigned to node " + firstDataNodeName);
-        safeAwait(verifyAssignmentToFirstNodeListener);
+        logger.info("--> Waiting for all [" + randomNumberOfShards + "] shards to be assigned to node " + firstDataNodeName);
+        try {
+            safeAwait(verifyAssignmentToFirstNodeListener);
+        } catch (AssertionError error) {
+            dumpClusterState();
+            throw error;
+        }
```

Comment:
This try-catch is the actual change. The others are extact an reusable method and some minor tweaks.